api/lib/gateways/Academy-Benefits/Search.js

```diff
@@ -1,82 +1,90 @@
 
-  const getDbRecords = async queryParams => {
-    try {
-      return await searchDb.execute(queryParams);
-    } catch (err) {
-      logger.error(
-        `Error searching customers in Academy-Benefits: ${err}`,
-        err
-      );
-      return [];
+  const search = async queryParams => {
+    let response = await callApi(queryParams);
+    let claimants = response.claimants;
+
+    while (response.nextCursor) {
+      response = await callApi(queryParams, response.nextCursor);
+      claimants = [...claimants, ...response.claimants];
     }
+    return processRecords(claimants);
   };
 
   const getApiRecords = async queryParams => {
     try {
-      return await searchAPI.execute(queryParams);
+      return await search(queryParams);
     } catch (err) {
       logger.error(
-        `Error searching customers in Academy-Benefits API: ${err}`,
+        `Error searching customers in Academy-Benefits API: ${err.error}`,
         err
       );
       return [];
     }
   };
 
-  const checkRecordsAreIdentical = (dbRecords, apiRecords) => {
-    if (sameResults(dbRecords, apiRecords)) {
-      logger.log(
-        'Academy records retrieved from the API and the DB are identical'
-      );
-    } else {
-      logger.log('Academy API and DB have returned different records');
-      logger.log({ 'DB records': dbRecords });
-      logger.log({ 'API records': apiRecords });
-    }
-  };
-
-  const sameResults = (dbRecords, apiRecords) => {
-    if (dbRecords.length !== apiRecords.length) return false;
-    let equal = true;
-    for (const record of dbRecords) {
-      if (!apiRecords.find(r => recordEquality(record, r))) {
-        equal = false;
-        break;
+  const callApi = async (queryParams, cursor) => {
+    return await rp(`${baseUrl}/api/v1/claimants`, {
+      method: 'GET',
+      headers: {
+        'X-API-Key': apiKey
+      },
+      json: true,
+      qs: {
+        first_name: queryParams.firstName,
+        last_name: queryParams.lastName,
+        cursor: cursor,
+        limit: 100
       }
-    }
-    return equal;
+    });
   };
 
-  const recordEquality = (record1, record2) => {
-    return (
-      record1.id === record2.id &&
-      stringDataEquality(record1.firstName, record2.firstName) &&
-      stringDataEquality(record1.lastName, record2.lastName) &&
-      stringDataEquality(record1.nino, record2.nino) &&
-      stringDataEquality(record1.postcode, record2.postcode) &&
-      (record1.dob === record2.dob ||
-        moment(record1.dob).diff(moment(record2.dob), 'days') < 1)
-    );
+  const validateIds = record => {
+    return record.claimId && record.checkDigit && record.personRef;
   };
 
-  const stringDataEquality = (string1, string2) => {
-    if (string1 == string2) return true;
-    if (!string1 || !string2) return false;
-    return string1.toLowerCase().trim() === string2.toLowerCase().trim();
+  const processRecords = records => {
+    return records
+      .filter(record => validateIds(record))
```

Comment:
Single view uses these 3 "ID"'s to form a new ID. I guess without them it's likely that the record is invalid. It might be that we should also be filtering out records like this in the API, but would definitely need to confirm this with a domain expert.